### PR TITLE
Consoles component API  FE

### PR DESCRIPTION
**Pull request by MShiers7**

### CODE REVIEW

**Review by KayPaint — APPROVED** [review]

File: src/api/admin.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  title,
+  description,
+  year,
+  price,
+}) => {
+  try {
+    const response = await fetch(`${BASEURL}/consoles`, {
+      method: "POST",
+      headers: {
+        "Content-type": "Application/json",
```
Comment (Remics89) [inline src/api/admin.js]: Im not sure this is an issue, but case sensitivity on "Application/json" could possibly present a problem vs. application/json

File: src/components/index.js [hunk 2]
```diff
@@ -1 +1,2 @@
-export { default as App } from './App';
\ No newline at end of file
+export { default as App } from './App';
+export { default as consoles } from './consoles';
```
Comment (jhoolwerf) [inline src/components/index.js]: change consoles to uppercase

File: src/api/admin.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+export const fetchUpdateConsoleListing = async ({
+  title,
+  description,
+  price,
+}) => {
+  try {
+    const response = await fetch(`${BASEURL}/consoles/${consoleId}`, {
+      method: "PATCH",
+      headers: {
+        "Content-type": "Application/json",
```
Comment (Remics89) [inline src/api/admin.js]: possibly same here

File: src/api/admin.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    console.error(error);
+  }
+};
+
+export const fetchDeleteConsoleListing = async () => {
+  try {
+    const response = await fetch(`${BASEURL}/consoles/${consoleId}`, {
+      method: "DELETE",
+      headers: {
+        "Content-type": "Application/json",
```
Comment (Remics89) [inline src/api/admin.js]: and here

File: src/api/consoles.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import BASEURL from "./index";
+
+export const fetchAllConsoles = async () => {
+    try {
+        const response = await fetch(`${BASEURL}/consoles`, {
+            headers: {
+                'Content-Type': 'Application/json'
```
Comment (Remics89) [inline src/api/consoles.js]: here

File: src/api/consoles.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        console.error("An error occured while attempting to fetch consoles", error);
+    }
+}
+
+
+export const fetchConsoleById = async (consoleId) => {
+    try {
+        const response = await fetch(`${BASEURL}/consoles/${consoleId}`, {
+            headers: {
+                'Content-Type': 'Application/json'
```
Comment (Remics89) [inline src/api/consoles.js]: here

File: src/api/consoles.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        console.error("An error occured while attempting to fetch the console by ID", error);
+    }
+}
+
+
+
+export const fetchConsolesByYear = async (year) => {
+    try {
+        const response = await fetch(`${BASEURL}/consoles/year/${year}`);
+        const results = await response.json();
```
Comment (Remics89) [inline src/api/consoles.js]: headers missing here

**Review by Remics89 — CHANGES_REQUESTED** [review]

**Review by Remics89 — APPROVED** [review]

**Pull request merged**